### PR TITLE
chore: remove old mutagen sync method coda paths

### PR DESCRIPTION
**Pull request by 10ko**
**What this PR does / why we need it**:

This PR removes any reference to the old `mutagen` sync mode and removes the `GARDEN_ENABLE_NEW_SYNC` environment variable.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by 10ko** [comment]
@vvagaytsev I left the mention of switching between old and new sync modes in the docs but I am not sure if that's still relevant (at least parts of it). Could you take a look at that and let me know?

File: core/src/graph/config-graph.ts [hunk 1]
```diff
@@ -449,6 +449,7 @@ export abstract class BaseConfigGraph<
       }
     }
 
+    // @ts-ignore
```
Comment (vvagaytsev) [inline core/src/graph/config-graph.ts:452]: It looks like this is not related to the sync mode, was that done intentionally?

Comment (vvagaytsev) [reply core/src/graph/config-graph.ts]: I deleted this, otherwise the linter was failing.

**Comment by twelvemo** [comment]
sorry @10ko i accidentally closed this while i actually wanted to close another PR

File: docs/guides/code-synchronization.md [hunk 2]
```diff
@@ -322,30 +322,61 @@ garden util mutagen sync list
...
+
+When downgrading, to stop the new sync daemon and to deploy with old sync mode, you need to run the following commands from the project root directory:
+
+```sh
+# If you are downgrading to Garden >= 0.13.33
+garden util mutagen daemon stop
+garden self-update <your-preferred-version>
+garden deploy --sync
+
+# If you are downgrading to Garden >= 0.13.22 and <=0.13.32 and want to use the old sync machinery
```
Comment (vvagaytsev) [inline docs/guides/code-synchronization.md:363]:
```suggestion
# If you are downgrading to Garden >= 0.13.26 and <=0.13.32 and want to use the old sync machinery
```

**Review by vvagaytsev — APPROVED** [review]
